sw/device/tests/entropy_src_fw_observe_many_contiguous.c

```diff
@@ -136,13 +150,21 @@ status_t firmware_override_observe(
     // Collect samples from the the observe FIFO.
     uint32_t words_to_read = nr_sample_words;
     uint32_t *sample_buffer_ptr = sample_buffer;
+    // Clear POSTHT_ENTROPY_DROP_ALERT bit.
+    TRY(dif_entropy_src_clear_recoverable_alerts(
+        &entropy_src,
+        ENTROPY_SRC_RECOV_ALERT_STS_POSTHT_ENTROPY_DROP_ALERT_BIT));
     // Drain FIFO to make sure we get contiguous samples.
     LOG_INFO("drain observe FIFO overflow...");
     TRY(entropy_testutils_drain_observe_fifo(&entropy_src));
```

Comment:
Should the alert clearing be done after draining the FIFO? I think potentially one could be unlucky and the alert could trigger again between drain the FIFO and clearing the error?

Reply:
Good question! Let me try to explain: This alert is independent of the observe FIFO. If the observe FIFO is full, we don't add further data to the observe FIFO but it continues to flow downstream into the conditioner (we're running Firmware Override: Observe mode here). In case of conditioner back pressure (it's very very unlikely to see this under normal operating conditions), entropy is being dropped at the input of the postht FIFO which is the FIFO before the observe FIFO. Draining the observe FIFO has zero impact on whether we're going to see such an alert.

If we first drained the observe FIFO and then cleared the alert bit, it could be that between draining the FIFO and clearing the alert bit, the conditioner exhibits back pressure and we drop a sample at the postht input. Since we already drained the observe FIFO and restarted collection, the collected entropy is no longer contiguous. Which is the issue we need to avoid using this alert bit.

Reply:
Thanks for the explanation

Reply:
No worries, thanks for your review!